### PR TITLE
[Jetpack Social] Implement UI states in Jetpack Social container (Post Settings)

### PR DESCRIPTION
**Pull request by RenanLukas**
Part of #18571 

Testing the following states with real data is only possible with the changes in https://github.com/wordpress-mobile/WordPress-Android/pull/18782.

No connections 
image

With connections and share limit enabled
image

With connections and share limit disabled should only display the list of connections and the message container.

To test:
Nothing to test except making sure `EditPostSettingsJetpackSocialNoConnectionsPreview` and `EditPostSettingsJetpackSocialContainerPreview` are still working as expected.

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and unit testing

3. What automated tests I added (or what prevented me from doing so)
`GetJetpackSocialShareLimitStatusUseCaseTest.kt`

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:

- [x] Portrait and landscape orientations.
- [x] Light and dark modes.
- [x] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [x] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostPublishSettingsViewModel.kt [hunk 1]
```diff
@@ -68,4 +79,43 @@ class EditPostPublishSettingsViewModel @Inject constructor(
     }
 
     fun getAuthorIndex(authorId: Long) = authors.value?.indexOfFirst { it.personID == authorId } ?: -1
+
+    fun onJetpackSocialConnectProfilesClick() {
+        // TODO
```
Comment (RenanLukas) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostPublishSettingsViewModel.kt:84]: The TODOs in this class will be handled in another PR.

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsJetpackSocialContainerView.kt [hunk 2]
```diff
@@ -3,48 +3,53 @@ package org.wordpress.android.ui.posts
...
-                    remainingSharesMessage = it.remainingSharesMessage,
-                    subscribeButtonLabel = it.subscribeButtonLabel,
-                    onSubscribeClick = it.onSubscribeClick,
-                    connectProfilesButtonLabel = it.connectProfilesButtonLabel,
-                    onConnectProfilesCLick = it.onConnectProfilesCLick,
-                )
+            with(uiState.value) {
+                when (this) {
+                    is Loading -> {
+                        /*TODO*/
```
Comment (RenanLukas) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsJetpackSocialContainerView.kt:33]: This TODO will be handled in another PR.

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostPublishSettingsViewModel.kt [hunk 3]
```diff
@@ -68,4 +79,43 @@ class EditPostPublishSettingsViewModel @Inject constructor(
...
+
+    fun onJetpackSocialConnectionClick() {
+        // TODO
+    }
+
+    fun onJetpackSocialMessageClick() {
+        // TODO
+    }
+
+    fun onJetpackSocialSuscribeClick() {
```
Comment (RenanLukas) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostPublishSettingsViewModel.kt:95]: Typo fixed in another PR.

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostPublishSettingsViewModel.kt [hunk 4]
```diff
@@ -68,4 +79,43 @@ class EditPostPublishSettingsViewModel @Inject constructor(
...
+            val shareMessage: String,
+            val remainingSharesMessage: String,
+            val subscribeButtonLabel: String,
+            val onSubscribeClick: () -> Unit,
+        ) : JetpackSocialUiState()
+
+        data class NoConnections(
+            val trainOfIconsModels: List<TrainOfIconsModel>,
+            val connectProfilesButtonLabel: String,
+            val onConnectProfilesCLick: () -> Unit,
```
Comment (RenanLukas) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostPublishSettingsViewModel.kt:114]: Capitalized "L" (`CLick`) fixed in another PR.

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostPublishSettingsViewModel.kt [hunk 5]
```diff
@@ -34,6 +39,12 @@ class EditPostPublishSettingsViewModel @Inject constructor(
     // Used for combining fetched users
     private val fetchedAuthors = mutableListOf<Person>()
 
+    private val _jetpackSocialuiState = MutableStateFlow<JetpackSocialUiState>(JetpackSocialUiState.Loading)
```
Comment (RenanLukas) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostPublishSettingsViewModel.kt:42]: Lowercase "u" (`_jetpackSocialuiState`) fixed in another PR.

**Review by develric — APPROVED** [review]
Thanks for the PR @RenanLukas, also for adding unit tests and trying to break the changes down into smaller PRs 🙇. I checked the previews for `EditPostSettingsJetpackSocialNoConnectionsPreview` and `EditPostSettingsJetpackSocialContainerPreview` and components look consistent in both light and dark mode 👍. I'm approving and merging this one :shipit: 

PS: np for future PRs as we were discussing in DM; about the typos you identified and corrected in subsequent PRs let's in general correct them directly in the original PR and then merge the small changes into the chained PRs, so no need to place notes for that 🙇
